### PR TITLE
Connect api

### PR DESCRIPTION
**Pull request by IfzaRasool**
**Connect to API**
- Add redux-thunk to the project.
- Use fetch for making API calls.
- Fetch book list from Bookstore API on Books page load.
- Add an Action Creator, that returns a function.
- Fetch a list of books.
- Dispatch another action (add an action type) that will pass the API response to the reducer and update the state.
- Refactor the add book and remove book features to persist your changes in the server.
- Change the existing Action Creators, so they return functions.
- Make changes by using API.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -41,15 +27,59 @@ const bookReducer = (state = initialState, action) => {
...
+  const result = await fetch(`${BASE_URL}/${id}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  if (result) {
+    dispatch(removeitem(id));
+  }
+};
```
Comment (BenMukebo) [inline src/redux/books/books.js:84]: - [ ] Great job so far! 👏🏻  However, I noticed that your books are not permanently deleted and they still show up when I reload the page. Please correct this

**Review by BenMukebo — APPROVED** [review]
### Hi @IfzaRasool,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻

- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- Professional README ✔️ 
- You have followed  JavaScript best practices. ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (BenMukebo) [inline src/redux/books/books.js]: - [x] Great job so far! 👏🏻  However, I noticed that your books are not permanently deleted and they still show up when I reload the page. Please correct this.

**Review by BenMukebo — CHANGES_REQUESTED** [review]

**Review by ShadyShawkat — APPROVED** [review]
## Hi @IfzaRasool, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**